Snoop.Core/Infrastructure/TreeExporter.cs

```diff
@@ -42,11 +44,46 @@ public bool Recurse
         get { return (bool)this.GetValue(RecurseProperty); }
         set { this.SetValue(RecurseProperty, value); }
     }
+
+    public bool ExportXamlStyle { get; set; }
+
+    public bool IncludeDefaultEmptyValues { get; set; }
+
+    public bool IncludeTypenameOnlyValues { get; set; }
+
+    public bool IncludeSystemCollectionNamespaceValues { get; set; }
+
+    public bool RoundDecimals { get; set; }
 }
 
 public class XMLTreeExporter
 {
-    public void Export(TreeItem treeItem, TextWriter textWriter, PropertyFilter? filter, bool recurse)
+    private static readonly Dictionary<Type, object?> defaultValueCache = new();
+
+    private static object? GetDefaultValue(Type type)
+    {
+        if (!defaultValueCache.TryGetValue(type, out var defaultValue))
+        {
+            defaultValue = Activator.CreateInstance(type);
```

Comment:
We shouldn't create unknown types.
It might be everything and it might be expensive.

Reply:
Uh, sorry. Didn't notice the restriction for value types. Then it should be ok and there is no need to restrict it further.